internal/entities/quote/common.go

```diff
@@ -44,6 +46,38 @@ func ValidateQuoteHash(hash string) error {
 }
 
 func CalculateCallFee(amount *entities.Wei, config PegConfiguration) *entities.Wei {
-	// TODO implement in GBI-2528
-	return entities.NewWei(100000000000)
+	result := new(entities.Wei)
+
+	percentageFee := calculatePercentageFee(amount, config.GetFeePercentage())
+	result.Add(percentageFee, config.GetFixedFee())
+
+	log.Debugf("Percentage fee: %v%% of %v = %v", config.GetFeePercentage(), amount, percentageFee)
+	log.Debugf("Fixed fee: %v", config.GetFixedFee())
+	log.Debugf("Call fee: %v + %v = %v", percentageFee, config.GetFixedFee(), result)
+	return result
+}
+
+func calculatePercentageFee(amount *entities.Wei, percentage *utils.BigFloat) *entities.Wei {
+	const scale = 1000 // the scale needs to have at least as many zeros as the amount of decimals we want to support in the percentage
+	amountAsRat := new(big.Rat).SetInt(amount.AsBigInt())
+	floatPercentage, _ := percentage.Native().Float64()
+
+	percentageAsFraction := new(big.Rat).SetFrac(
+		big.NewInt(int64(floatPercentage*scale)), // Scale to avoid precision loss
+		big.NewInt(100*scale),
+	)
+	percentageFee := new(big.Rat).Mul(amountAsRat, percentageAsFraction)
```

Comment:
There is any check for integer overflow on this multiplication?

Reply:
no, the big package allocates memory dynamically instead of using a fixed memory size like the native types https://pkg.go.dev/math/big#NewRat